### PR TITLE
.github: combine workflows, use refactored actions, align execution

### PR DESCRIPTION
**Pull request by MSevey**
This PR restructures the CI to be more unified, with (more or less) one single file that captures the execution. 

With this format, it drive the process of ensuring all linting and testing is passing before key actions like releases or docker builds are generated.

### CODE REVIEW

File: .github/workflows/ci_release.yml
Comment (MSevey) [inline .github/workflows/ci_release.yml:74]: @jbowen93 leaving a call out here for your docker build updates, just fyi

File: .github/workflows/clean-artifacts.yml
Comment (MSevey) [inline .github/workflows/clean-artifacts.yml:1]: Is this file doing something different than this repo setting? if not we can remove this file and just update the repo setting. 

Screen Shot 2022-10-18 at 2 58 49 PM

File: .github/workflows/linkchecker.yml
Comment (MSevey) [inline .github/workflows/linkchecker.yml:1]: see https://github.com/celestiaorg/.github/pull/3

File: .github/workflows/stale.yml
Comment (MSevey) [inline .github/workflows/stale.yml:1]: see https://github.com/celestiaorg/.github/pull/3

File: .github/workflows/tag.yml
Comment (evan-forbes) [reply .github/workflows/tag.yml:11]: yeah, I think we can remove it

File: .github/ISSUE_TEMPLATE/bug-report.yml
Comment (SweeXordious) [inline .github/ISSUE_TEMPLATE/bug-report.yml:11]:
```suggestion
        core modules and if it's eligible for a bug bounty on `SECURITY.md`. Bugs
```

File: .github/workflows/test.yml
Comment (SweeXordious) [inline .github/workflows/test.yml:143]: [Non Blocking][Suggestion]
Same as above for formatting correctly the embedded shell script

File: .github/workflows/proto.yml
Comment (SweeXordious) [inline .github/workflows/proto.yml:22]: [Non-Blocking][Suggestion]
Instead of disabling yaml lint, probably fix the indentation like mentionned here: https://stackoverflow.com/a/69825610/11529598

File: .github/workflows/test.yml
Comment (SweeXordious) [inline .github/workflows/test.yml:16]: [Non-Blocking][Suggestion]
Doesn't it make sense to increase the line length in the configuration since multiple yaml files are having this issue, instead of having to disable it everytime?

**Review by SweeXordious — COMMENTED** [review]
LGTM, but I will defer to the other reviewers, I don't have enough knowledge on this.
Left some non blocking questions/suggestions

File: .github/workflows/ci_release.yml
Comment (evan-forbes) [reply .github/workflows/ci_release.yml:72]: [non blocking][questions]
can we create an issue for this? I don't think the ephemeral cluster is using the docker builds as we just added in #881 and #882, should they be using the same? ref https://github.com/celestiaorg/.github/pull/4

The ones we just added should build when pushing matching tags, so I don't think we have to do anything there. 

possibly stupid question, will we be using this https://github.com/celestiaorg/.github/blob/main/.github/actions/docker-publish/action.yml ?

**Review by evan-forbes — COMMENTED** [review]
can we also delete https://github.com/celestiaorg/celestia-app/blob/sevey/ci-release/.github/workflows/stale.yml?

File: .github/workflows/test.yml
Comment (MSevey) [reply .github/workflows/test.yml:143]: The issue is line length, and I don't believe I can break up the lines that are violating the rule, which is why I did it this way.

File: .github/workflows/proto.yml
Comment (MSevey) [reply .github/workflows/proto.yml:22]: That is a different issue, that stackoverflow is asking about invalid workflow, which is Github linting the workflow, not `yamllint`.

Yamllint was failing for line length. I choose to disable the lint for these blocks because I didn't want to mess with the code block for something basic like line length.

Comment (MSevey) [reply .github/workflows/proto.yml:22]: same response.

File: .github/workflows/test.yml
Comment (MSevey) [reply .github/workflows/test.yml:16]: not imo, the line length is nice for readability.  Where possible I simply fixed the line to comply. There were a few places that I didn't want to touch the code to avoid opening a can of worms.

**Comment by MSevey** [comment]
> possibly stupid question, will we be using this https://github.com/celestiaorg/.github/blob/main/.github/actions/docker-publish/action.yml ?

Not a stupid question :-) 

That was a docker action I've used in the past, but it is set to push to DockerHub.  Looking at the docker build workflows in the various Celestia repos, it looks like we are pushing to the github docker image repository. So we could use it, but we would need to update it, so it is probably easiest to just replace it with the ones we are using. 

I'll link to this response in the issue I created as well.

File: .github/workflows/clean-artifacts.yml
Comment (MSevey) [reply .github/workflows/clean-artifacts.yml:1]: @evan-forbes thought on removing the workflow and just using this repo setting?  Do you remember why this workflow was added?

File: Dockerfile
Comment (SweeXordious) [inline Dockerfile:3]: [Non Blocking][Question]
> By default apk tries to select a set of packages so that all dependencies are satisfied. This means that some packages can be held back in an older version if there is direct or implicit dependencies to the older version.
https://man.archlinux.org/man/apk-add.8.en

I guess it's better not to force `latest` as it might not be compatible with the already installed packages in the base image. What do you think?

**Review by SweeXordious — COMMENTED** [review]
Left questions concerning the dockerfile versions

**Review by evan-forbes — APPROVED** [review]
pre-approving, but I think the questions Rachid asked are worth answering before merging (so I disabled auto merge this time)

File: Dockerfile
Comment (MSevey) [reply Dockerfile:3]: So these were `WARNINGS` about not specifying a version. In general it is best practice to use specific versions but `apk` is a nice tool.

If we merge this https://github.com/celestiaorg/.github/pull/8 I can revert these changes.

File: Dockerfile
Comment (MSevey) [reply Dockerfile:9]: same answer as above.

File: Dockerfile
Comment (SweeXordious) [reply Dockerfile:3]: Understandable. I am fine with whatever you prefer :cat:

**Review by SweeXordious — DISMISSED** [review]

File: Dockerfile
Comment (MSevey) [reply Dockerfile]: Went with ignoring this rule inline in favor of `apk` compatibility.

File: Dockerfile
Comment (MSevey) [reply Dockerfile:10]: Added dependabot docker to handle this.

File: .github/workflows/clean-artifacts.yml
Comment (evan-forbes) [reply .github/workflows/clean-artifacts.yml:1]: yeah, I think we can delete this

Comment (evan-forbes) [reply .github/workflows/clean-artifacts.yml:1]: It was likely copied over from the sdk or default from startport

**Review by evan-forbes — APPROVED** [review]

**Review by SweeXordious — APPROVED** [review]

**Pull request merged**
